src/c_api.cpp

```diff
@@ -175,8 +175,11 @@ class Booster {
     } else {
       is_raw_score = false;
     }
+    auto param = ConfigBase::Str2Map(parameter);
```

Comment:
We won't call the c_api one-time for one sample, right ? 
The  prediction is called one-time for many instances.